### ISSUE TITLE
Better JSON input field

### ISSUE THREAD

**Issue by ChromeGG** [dbcodeio/public#107]
Hi Mike,

Inline edits for JSON fields have quite poor UX. 
![Image](https://github.com/user-attachments/assets/98aee126-caef-4cf6-84d7-0f778b913be6)

Can we have something more handy to use? 😄 

Cheers!

**Comment by mikeburgh** [comment]
Agreed! I implemented the first half of my idea which is the icon at the end of the cell, taking you to full vscode editor, and all its goodness.

I just need to manage the update back from their with a virtual file system so changes are reflected and can be applied.

Do you think that UX would work, or are you looking for something in the table itself ? I have avoided that so far as you cannot get the vscode editor in there, so any editor surface is going to feel less than what you expect…. well at least that's my thinking

**Comment by ChromeGG** [comment]
I didn’t realize that this could be the flow. I thought that this icon was supposed to be something like 'Open in new tab to easily copy/preview.' I didn’t associate it with editing. 😅

Yeah, so this UX would work, I think. Maybe we can change the icon to a pencil and update the hover tooltip to something like 'Edit in a new tab' or similar.

Thanks!

**Comment by ChromeGG** [comment]
Yeah, I'll definitely try it, thanks!

**Comment by mikeburgh** [comment]
I think I have cracked this.. just pushed a potential solution in 1.14.29. If the column is json/xml and you edit it.. will automatically open a vscode "like" editor.. no fancy extensions from vscode.. but gives you the bulk.<br><br>Otherwise if it's a normal field, you still have that open in editor icon, which will then pop up the same editor.<br><br>I think it's a big improvement keeping it in the table.. rather than a new tab, only downside is I cannot get the full vscode editor and extensions you have installed into the table.. so hence the "like"

**Comment by ChromeGG** [comment]
<img width="457" height="152" alt="Image" src="https://github.com/user-attachments/assets/48dc59fe-fbd1-491b-943f-45d9373ba453" />

@mikeburgh not sure if it's related, but now I see `[object Object]` 😅

**Comment by mikeburgh** [comment]
This is Postgres right ? It's a bug when you first open a table from cache… I think I have a solution, but if you re open the table it should be good.

issue is Postgres needs type data from the db, and the table loads before that type data is fetched (to keep it fast) if the data was already cached…. But the pg driver does not currently read that cached data..

**Comment by mikeburgh** [comment]
This bug should be fixed now as well, closing this.

**Comment by ChromeGG** [comment]
Hi @mikeburgh 

Great job, now it looks and works I as expected! Thx.

There is one more small nitpick, I cannot close it by clicking outside the modal:

<img width="1424" height="622" alt="Image" src="https://github.com/user-attachments/assets/049381a8-32bc-4995-9056-4fba4607686e" />

Not a big deal, but quite unintuitive and I believe easy to fix 😄

**Comment by mikeburgh** [comment]
Fiddly one to get right, but should be good now in 1.16.9 which has been released.
